test/pylib/manager_client.py

```diff
@@ -109,8 +112,18 @@ def _driver_update(self) -> None:
             logger.debug("refresh driver node list")
             self.ccluster.control_connection.refresh_node_list_and_token_map()
 
-    async def before_test(self, test_case_name: str) -> None:
-        """Before a test starts check if cluster needs cycling and update driver connection"""
+    async def before_test(self, test_case_name: str, test_log: Path) -> None:
+        # Pytest uses 'test' handler to for output logs during tests.
+        # So we are adding a handler for it here to write all logs to test_log file. Since logger instance is singleton,
+        # pytest will use this one instead of creating a new one
+        test_logger = logging.getLogger('test')
+        self.test_log_fh = logging.FileHandler(test_log, mode='w+')
+        # to get the formatter with a timestamp we need to add explicitly here for our handler
+        # to make it consistent with the root logger formatter we extract it and applying to the handler
+        self.test_log_fh.setFormatter(logging.getLogger().handlers[0].formatter)
```

Comment:
We have formatter set on the root logger to have timestamps and suffix of the test suite, that's why I'm getting this formatter and applying it here. 
```logger.getLogger()``` returns the root logger and while ```test_logger``` is different, since I'm explicitly getting it.

Reply:
Ok. (Could be nice to have this explained in the comment though.)

Reply:
There is already a comment on why we are getting formatter from the root logger. I'll try to rephrase it since it seems not clear.